### PR TITLE
[Prometheus] Mesh setup for AlertManager StatefulSet

### PR DESCRIPTION
**Pull request by danielfm**
#### What this PR does / why we need it:
In order to set up the mesh cluster in Alertmanager, we must specify the `--cluster.peer` argument for each peer, but this can't be done via the `alertmanager.extraArgs` parameter because we cannot set the same YAML key more than once:

```yaml
# ...
alertmanager:
  extraArgs:
    cluster.listen-address: 0.0.0.0:6783
    cluster.peer: prometheus-alertmanager-0.prometheus-alertmanager-headless:6783

    # Will overwrite the previous arg
    cluster.peer: prometheus-alertmanager-1.prometheus-alertmanager-headless:6783
```

So this PR automates the mesh cluster configuration by changing the Alertmanager Statefulset to automatically set up the `--cluster.listen-address` and `--cluster.peer` flags for each Statefulset member via the headless service, if enabled.

#### Which issue this PR fixes
*(optional, in `fixes #<issue number>(, fixes #<issue_number>, ...)` format, will close that issue when PR gets merged)*
  - n/a

#### Checklist
[Place an '[x]' (no spaces) in all applicable fields. Please remove unrelated fields.]
- [x] [DCO](https://www.helm.sh/blog/helm-dco/index.html) signed
- [x] Chart Version bumped

### CODE REVIEW

File: stable/prometheus/templates/alertmanager-statefulset.yaml [hunk 1]
```diff
@@ -58,6 +58,12 @@ spec:
             - --config.file=/etc/config/alertmanager.yml
             - --storage.path={{ .Values.alertmanager.persistentVolume.mountPath }}
             - --cluster.advertise-address=$(POD_IP):6783
+          {{- if .Values.alertmanager.statefulSet.headless.enableMeshPeer }}
```
Comment (gianrubio) [inline stable/prometheus/templates/alertmanager-statefulset.yaml:53]: please add this variables on Readme.md

Comment (danielfm) [reply stable/prometheus/templates/alertmanager-statefulset.yaml:53]: I did not introduce any new variables. These are already listed on the README.

**Comment by danielfm** [comment]
Squashed and rebased.

/retest

**Pull request merged**